examples/addsvc/client/grpc/factory.go

```diff
@@ -7,33 +7,42 @@ import (
 
 	"github.com/go-kit/kit/endpoint"
 	"github.com/go-kit/kit/examples/addsvc/pb"
+	"github.com/go-kit/kit/loadbalancer"
+	kitot "github.com/go-kit/kit/tracing/opentracing"
 	grpctransport "github.com/go-kit/kit/transport/grpc"
+	"github.com/opentracing/opentracing-go"
 )
 
 // SumEndpointFactory transforms GRPC host:port strings into Endpoints that call the Sum method on a GRPC server
 // at that address.
-func SumEndpointFactory(instance string) (endpoint.Endpoint, io.Closer, error) {
-	cc, err := grpc.Dial(instance, grpc.WithInsecure())
-	return grpctransport.NewClient(
-		cc,
-		"Add",
-		"Sum",
-		encodeSumRequest,
-		decodeSumResponse,
-		pb.SumReply{},
-	).Endpoint(), cc, err
+func NewSumEndpointFactory(tracer opentracing.Tracer) loadbalancer.Factory {
```

Comment:
btw, there is an `opentracing.GlobalTracer()` singleton; if we embrace it, the indirection here would not be necessary.